orca-mahe/src/test/groovy/com/netflix/spinnaker/orca/mahe/tasks/CreatePropertiesTaskSpec.groovy

```diff
@@ -138,6 +138,56 @@ class CreatePropertiesTaskSpec extends Specification {
     }
   }
 
+  def "delete single persisted properties"() {
```

Comment:
Might be nice if things were refactored slightly such that we didn't need a 50 line test case, is this possible?

I don't quite know the context behind  deleting properties in the `CreatePropertiesTask`, but at a minimum you could extract the bit that's iterating through properties and upserting/deleting into it's own method that could be tested directly.

Or maybe you don't need such a verbose `given` to test the task? It doesn't seem like you need two pipeline stages, the email address, etc?

Otherwise this LGTM.